spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -32,21 +33,37 @@
  * @since 1.0
  *
  */
-public class UnZipResultSplitter {
-
-	public List<Message<Object>> splitUnzippedMap(Map<String, Object> unzippedEntries) {
+public final class UnZipResultSplitter {
```

Comment:
@artembilan yes, absolutely agree with you.

But what kind of extension allows the class before my PR? Just simple subclassing? It's useless: subclass can't change any part of logic in solid `splitUnzippedMap` method. Only one way user can 'extend' it: call smth before or/and after `splitUnzippedMap`. But the same thing may be done only with delegation without any subclassing!

My changes provide splitting `splitUnzippedMap` into some pieces of algorithm user can easily and safely change in subclass. In my PR `splitUnzippedMap` is more like template method. That's was idea. Other good things about this splitting: decrease code complexity and increase code cleverness and, thus, maintainability.

But, again, my PR isn't about create better architecture for the class. It's only for bug fix and some obvious 'scout work' and refactoring.